### DISCUSSION TITLE
classes/class_navigationobstacle2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#130] [User-contributed notes]
# classes/class_navigationobstacle2d

Experimental: This class may be changed or removed in future versions. Inherits: Node2D< CanvasItem< Node< Object 2D obstacle used to affect navigation mesh baking or constrain velocities of avoida...

https://docs.godotengine.org/en/stable/classes/class_navigationobstacle2d.html

**Comment by ChickensioDev** [comment]
ur supposed to put the navigation obstacle 2ds as child of navigation region 2d in order for it to work. i spent over an hour searching everywhere. youtube, docs, and it wasnt mentioned and i just happened to find that out by chance.

**Reply by TavseKoder** [reply]
Thank you very much for this contribution. I was also looking at videos and forums and everything seemed too complicated.

**Comment by granthall1972** [comment]
I feel like radius should be renamed to avoidance_radius or similar. It isn’t obvious without reading the docs the radius only applies to avoidance and not used to block navigation or affect paths directly.

**Reply by Calinou** [reply]
Renaming the property would break compatibility with existing projects, so it can't be done now. That said, I've opened a PR to improve the description:

- https://github.com/godotengine/godot/pull/115317
